docs/modules/dimmer.md

```diff
@@ -0,0 +1,135 @@
+
+The above scheme is implemented in this module by dedicating ESP32's CPU1 entirely to this purpose... Phase dimming requires very accurate timing. Configuring timer interrupts in the "busy" CPU0 does not work properly, with FreeRTOS scheduler, WiFi and so on demanding their share of the CPU at random intervals, which would make dimmed lamps flicker.
+
+Once the dimmer module is started, by means of `dimmer.setup()`, a busy loop is launched on CPU1 that monitors zero-crossing signals from the dimmer and turns on/off the TRIAC at the appropriate time, with nanosecond precision.
+
+## Required SDK `menuconfig` changes
+To use this module, change the following in menuconfig (`make menuconfig`):
+
+* Enable *FreeRTOS in both cores* by unselecting *"Component Config/FreeRTOS/Run FreeRTOS only on first core"*. This will allow the module to use CPU1.
+* Unselect  *"Component Config/ESP32-specific/Also watch CPU1 tick interrupt"*
+* Unselect  *"Component Config/ESP32-specific/Watch CPU1 idle task"*
+
+The last two settings disable the reset watchdog for CPU1. This is necessary since CPU1 is purposefully going to be running an infinite loop.
+
+## Example
+
+```lua
+dimmer.setup(14)     -- configure pin 14 as zero-crossing detector
+dimmer.add(22)       -- TRIAC gate controlling lightbulb is connected to GPIO pin 22
+dimmer.setLevel(300) -- set brightness to 300‰ (30%)
+```
+
+
+## dimmer.setup()
+
+Initializes the dimmer module. This will start a task that continuously monitors for signal zero crossing and turns on/off the different dimmed pins at the right time.
+
+#### Syntax
+
+`dimmer.setup(zc_pin, [transition_speed])`
+
+#### Parameters
+
+* `zc_pin`: GPIO pin number where the zero crossing detector is connected. `dimmer.setup()` will configure this pin as input.
+* `transition_speed`: integer. Specifies a transition time or fade to be applied every time a brightness level is changed. It is defined as a per mille (‰) brightness delta per half mains cycle (10ms if 50Hz). For example, if set to 20, a light would go from zero to full brightness in 1000 / 20 * 10ms = 500ms 
+  *  Value must be between 1 and 1000 (instant change).
+  *  Defaults to a quick ~100ms fade.
+
+#### Returns
+
+Throws errors if CPU1 task cannot be started or `zc_pin` cannot be configured as input.
+
+It will reset if core 1 CPU is not active or their watchdogs are still enabled. See "Required SDK `menuconfig` changes" above.
+
+## dimmer.add()
+
+Adds the provided pin to the dimmer module. The pin will be configured as output.
+
+#### Syntax
+
+`dimmer.add(pin, [dimming_mode])`
+
+#### Parameters
+
+* `pin`: the GPIO pin to control.
+* `dimming_mode`: Dimming mode, either `dimmer.LEADING_EDGE` (default) or `dimmer.TRAILING_EDGE`, depending on the type of load or lightbulb.
+
+#### Returns
+
+`dimmer.add()` will throw an error if the pin was already added or if an incorrect GPIO pin is provided.
+
+## dimmer.remove()
+
+Removes the given pin from the dimmer module
```

Comment:
Is it intentional that this does not set the level back to 0 i.e. the light remains on? Intuitively I was expecting that.